instrumentation/src/javatests/org/oppia/android/instrumentation/topic/TopicTest.kt

```diff
@@ -0,0 +1,100 @@
+package org.oppia.android.instrumentation.topic
+
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.uiautomator.UiDevice
+import com.google.common.truth.Truth.assertThat
+import org.junit.Before
+import org.junit.Test
+import org.oppia.android.instrumentation.testing.EndToEndTestHelper.findObjectByRes
+import org.oppia.android.instrumentation.testing.EndToEndTestHelper.findObjectByText
+import org.oppia.android.instrumentation.testing.EndToEndTestHelper.scrollRecyclerViewTextIntoView
+import org.oppia.android.instrumentation.testing.EndToEndTestHelper.startOppiaFromScratch
+import org.oppia.android.instrumentation.testing.EndToEndTestHelper.waitForRes
+import org.oppia.android.instrumentation.testing.EndToEndTestHelper.waitForText
+
+/** Tests for the topic viewer screen. */
+class TopicTest {
+  private lateinit var device: UiDevice
+
+  @Before
+  fun setUp() {
+    // Initialize UiDevice instance
+    device = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+    device.startOppiaFromScratch()
+  }
+
+  @Test
+  fun testTopicActivity_exploreInfoFragment_finishesSuccessfully() {
```

Comment:
'Fragment' is too implementation specific. This should be tied only to user behaviors, e.g. ``testTopicViewer_exploreInfoFragment_hasSummaryDetails`` or something similar. Ditto for the other tests.

Reply:
Done